### DISCUSSION TITLE
git-lfs storage

### DISCUSSION THREAD

**Discussion by greyltc** [requarks/wiki#7478] [Help / Questions]
Hi. The git storage backed isn't so great for non-text files like assets because git is revision control for text files. [git-lfs ](https://git-lfs.com/) gracefully extends git to work efficiently with binary blobs (like wiki assets). So wiki.js with a git-lfs storage backend seems like a really great match. Has there been any effort to get wiki.js's git storage backend working with git-lfs?

**Comment by burbschat** [comment]
Not perfect as I believe files can only be matched by filename and not file content, but it seems like when running in an environment where `git-lfs` is available and one prepares a repository containing a proper `.gitattributes` lfs is used for all matched files. I assume under the hood simply a `git` binary is called and as if lfs is set up, the usual `git add`, `git commit` and `git push` commands will lead to all matched files to be managed by lfs (I tried this, appears to work alright).
Note that the default docker image does not come with `git-lfs`. Have a look at [this](https://feedback.js.wiki/wiki/p/git-lfs-is-not-supported-out-of-box) related post.

Now it would be great if there was some way to have all files that git recognizes as binary to be tracked by lfs. Perhaps one could prepare some shell script that intercepts comitted files and checks if they are binary files, and if so tracks them with lfs. Then using the *Git Binary Path* setting to point to such a script might work.
